crates/relay/src/tests.rs

```diff
@@ -619,3 +640,67 @@ fn test_store_block_header_and_update_sla_fails_with_invalid() {
         );
     })
 }
+
+#[test]
+fn should_not_report_double_payment_with_identical_proof() {
+    run_test(|| {
+        assert_err!(
+            Relay::report_vault_double_payment(
+                Origin::signed(ALICE),
+                CAROL,
+                (vec![0u8; 32], vec![0u8; 32]),
+                (vec![0u8; 32], vec![0u8; 32]),
+            ),
+            TestError::DuplicateMerkleProof
+        );
+    })
+}
+
+#[test]
+fn should_not_report_double_payment_with_identical_tx() {
+    run_test(|| {
+        assert_err!(
+            Relay::report_vault_double_payment(
+                Origin::signed(ALICE),
+                CAROL,
+                (vec![0u8; 32], vec![1u8; 32]),
+                (vec![0u8; 32], vec![0u8; 32]),
+            ),
+            TestError::DuplicateTransaction
+        );
+    })
+}
+
+#[test]
+fn should_report_double_payment() {
```

Comment:
Can you check that a random person making a tx with the same op_return does not trigger theft? Just to catch any future regressions